pkg/integration_tests/oneagent_controller_integration_test.go

```diff
@@ -36,4 +37,33 @@ func TestReconcileOneAgent_ReconcileOnEmptyEnvironment(t *testing.T) {
 
 	assert.Equal(t, DefaultTestNamespace, dsActual.Namespace, "wrong namespace")
 	assert.Equal(t, oaName, dsActual.GetObjectMeta().GetName(), "wrong name")
+	assert.Equal(t, corev1.DNSClusterFirst, dsActual.Spec.Template.Spec.DNSPolicy) // default when empty.
+}
+
+func TestReconcileOneAgent_ReconcileOnEmptyEnvironmentAndDNSPolicy(t *testing.T) {
```

Comment:
While this test is great, can we also look into incorporating unit tests once again, now that we discovered the fake clientset can be used from controller-runtime?